### PR TITLE
Single Binary packing approach

### PR DESCRIPTION
**Pull request by guilhermebodin**

### CODE REVIEW

**Comment by guilhermebodin** [comment]
@raphasampaio @pedromxavier @joaquimg This is an attempt on packing the .hdr and .bin in a single .bin. I had a few errors with this approach but maybe it can be useful to discuss #63

**Comment by guilhermebodin** [comment]
@raphasampaio @joaquimg Acho que ja resolvi aqui.

File: src/OpenBinary/reader.jl [hunk 1]
```diff
@@ -58,26 +59,36 @@ function PSRI.open(
...
+        PATH_HDR = path * ".hdr"
+        PATH_BIN = path * ".bin"
+        if !isfile(PATH_BIN)
+            error("file not found: $PATH_BIN")
+        end
+        if !isfile(PATH_HDR)
+            error("file not found: $PATH_HDR")
+        end
+        ioh = open(PATH_HDR, "r")
+        seek(ioh, 0) # absolute position
```
Comment (joaquimg) [inline src/OpenBinary/reader.jl:89]: Only in else?

File: src/OpenBinary/reader.jl [hunk 2]
```diff
@@ -301,19 +322,19 @@ function PSRI.open(
...
 
-        io = open(PATH_BIN, "r")
+        io = io,
+        hs = hs,
     )
 
-    finalizer(ret) do x
-        if x.is_open
-            Base.close(x.io)
+    if !single_binary
```
Comment (joaquimg) [inline src/OpenBinary/reader.jl:329]: why not for single bin?

File: src/OpenBinary/reader.jl [hunk 3]
```diff
@@ -325,7 +346,11 @@ function PSRI.open(
     @assert last == position(ret.io)
 
     # go back to begning and initialize
-    seek(ret.io, 0)
+    if single_binary
```
Comment (joaquimg) [inline src/OpenBinary/reader.jl:349]: is this "if" needed?
ret.hs isnt already zero?
so `seek(ret.io, ret.hs)` works for both...

File: src/OpenBinary/reader.jl [hunk 4]
```diff
@@ -405,21 +430,25 @@ function _get_position(graf, t::Integer, s::Integer, b::Integer)
...
 function PSRI.next_registry(graf::Reader)
     if graf.stage_current == graf.stage_total &&
         graf.scenario_current == graf.scenario_total &&
         graf.block_current == graf.block_total
-        seek(graf.io, 0)
+        if graf.hs != 0 # single binary
+            seek(graf.io, graf.hs)
+        else
+            seek(graf.io, 0)
+        end
```
Comment (joaquimg) [inline src/OpenBinary/reader.jl:451]:
```suggestion
        seek(graf.io, graf.hs)
```

**Review by joaquimg — CHANGES_REQUESTED** [review]
nice! just some tweaks needed

File: src/OpenBinary/reader.jl (hunk 1 above)
Comment (guilhermebodin) [reply src/OpenBinary/reader.jl:89]: no need for this one actually

**Comment by pedromxavier** [comment]
@guilhermebodin I've just noticed that Classic's SingleBinary uses `.dat` extension. We're currently using `.bin`:

https://github.com/psrenergy/PSRClassesInterface.jl/blob/gb/single_bin/src/OpenBinary/reader.jl#L73
https://github.com/psrenergy/PSRClassesInterface.jl/blob/gb/single_bin/src/OpenBinary/writer.jl#L140

**Comment by raphasampaio** [comment]
@pedromxavier the user should be able to decide any extension that he wants, in the single binary, header+binary, and CSV modes

**Comment by guilhermebodin** [comment]
@raphasampaio so what do we do, fix it at `.bin` or add more options?

**Comment by raphasampaio** [comment]
It's better to fix it

**Comment by pedromxavier** [comment]
@raphasampaio @guilhermebodin  It looks good to have more options. In any case `.dat` should be the default behaviour to match ClassicInterface's IO for single bin.

**Comment by guilhermebodin** [comment]
Changing to .dat @pedromxavier have you tested the files against the original ones?

**Comment by pedromxavier** [comment]
@raphasampaio @guilhermebodin The last commits include:
- `hs::Int` removal from header contents.
- Added `skips::Union{Nothing, Vector{Tuple{Int, Int}}}` to track file cursos skips within the header in order to consider those chunks as garbage during binary comparison. Added also a parameter `store_skips::Bool` to `OpenBinary.Reader` to activate this feature.

**Pull request merged**